uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -203,6 +206,15 @@ define([
       };
 
       /**
+       * Manager selects the contact when applying leave on their behalf.
+       * Reinitialize all as the leave request is based on contact selection.
+       */
+      this.contactSelected = function () {
+        this.request.contact_id = this.contact.id;
+        this._init();
```

Comment:
Why do you need to reinitialize the entire controller when you change the user? You are reloading everything, statues, absence types, etc)